src/test/ui/lint/rfc-2383-lint-reason/expect_nested_lint_levels.rs

```diff
@@ -0,0 +1,29 @@
+// check-pass
+
+#![feature(lint_reasons)]
+
+#[expect(unused_mut, reason = "This should trigger because `unused_mut` has the level `allow`")]
+//~^ WARNING this lint expectation is unfulfilled [unfulfilled_lint_expectations]
+//~| NOTE `#[warn(unfulfilled_lint_expectations)]` on by default
+//~| NOTE This should trigger because `unused_mut` has the level `allow`
+mod foo {
+  fn bar() {
+    #[allow(unused_mut, reason = "v is unused")]
+    let mut v = 0;
+  }
+}
+
+#[expect(unused_mut, reason = "This should trigger because `unused_mut` has the level `warn`")]
+//~^ WARNING this lint expectation is unfulfilled [unfulfilled_lint_expectations]
+//~| NOTE This should trigger because `unused_mut` has the level `warn`
+mod oof {
+    #[warn(unused_mut, reason = "We no longer expect it in this scope")]
```

Comment:
Yes, this can be confusing at the start. The RFC stated:
> This RFC also adds the `expect(lint_name, reason = STRING)` lint attribute. The
> `expect` attribute uses the same lint-suppression mechanism that `allow` does,
> but will raise a new lint, `expectation_missing` (name open to change), when the
> lint it expects does not arrive.

The *"same lint-suppression mechanism that `allow` does"* part makes it sounds like this was expected to be like another lint level. The RFC discussion also hinted to the fact. I'll update the reason parameter to explain this better :upside_down_face: